### ISSUE TITLE
Hello

### ISSUE THREAD

**Issue by szj-github** [mrmq9/RNPCANet#1]
Hello, I'd like to ask you a question. Is RFF performed on a single image? thanks！

**Comment by mrmq9** [comment]
> Hello, I'd like to ask you a question. Is RFF performed on a single image? thanks！

Hi, I'm not sure if I understood your question correctly.
In RNPCANet, the RFF method is used to set the filters before learning the PCA filters. In RFF, as it is described in the paper, the weights are drawn from a normal distribution to approximate a kernel. After learning the parameters by RFF and PCA, the model is used for feature extraction for all the training samples to train a SVM.

**Comment by szj-github** [comment]
Thank you very much for your reply. My question may not be clear. I mean, does the RFF in the first stage of each time deal with a single image, or should I say batchsize=1?
![image](https://user-images.githubusercontent.com/54060024/171665194-b7e16348-ad35-4b0f-8178-55bf28163646.png)
![image](https://user-images.githubusercontent.com/54060024/171664726-ffeabd82-0222-4407-81b1-8959b582de1c.png)

**Comment by mrmq9** [comment]
> Thank you very much for your reply. My question may not be clear. I mean, does the RFF in the first stage of each time deal with a single image, or should I say batchsize=1? ![image](https://user-images.githubusercontent.com/54060024/171665194-b7e16348-ad35-4b0f-8178-55bf28163646.png) ![image](https://user-images.githubusercontent.com/54060024/171664726-ffeabd82-0222-4407-81b1-8959b582de1c.png)

No, you can use larger batch sizes than 1. The computations in utils/hist_hash.py  may need too much memory, so you should just ensure you have enough memory on your GPU for using larger batch sizes.
And also in the picture above, the set you indicated with the arrow is the set of extracted patches from **all** the training images.

**Comment by szj-github** [comment]
Thank you very much! That is, if there are 100 pictures, and each picture is divided into 10 patches (k1*k2), then N1 is equal to 100*10?

**Comment by szj-github** [comment]
Thank you very much! That is, if there are 100 pictures, and each picture is divided into 10 patches (k1*k2), then N1 is equal to 100*10?

**Comment by szj-github** [comment]
Thank you very much! That is, if there are 100 pictures, and each picture is divided into 10 patches (k1*k2), then n1 is equal to 1000?

**Comment by mrmq9** [comment]
> 

Yes, exactly.

**Comment by mrmq9** [comment]
> Thank you very much! That is, if there are 100 pictures, and each picture is divided into 10 patches (k1*k2), then n1 is equal to 1000?

Yes, exactly.

**Comment by szj-github** [comment]
Thank you very much, recently I am working on related principal component analysis network processing data, may cite your paper.

**Comment by szj-github** [comment]
> > Thank you very much! That is, if there are 100 pictures, and each picture is divided into 10 patches (k1*k2), then n1 is equal to 1000?
> 
> Yes, exactly.
Thank you very much, recently I am working on related principal component analysis network processing data, may cite your paper.

**Comment by mrmq9** [comment]
> > > Thank you very much! That is, if there are 100 pictures, and each picture is divided into 10 patches (k1*k2), then n1 is equal to 1000?
> > 
> > 
> > Yes, exactly.
> > Thank you very much, recently I am working on related principal component analysis network processing data, may cite your paper.

Great, good luck!

**Comment by szj-github** [comment]
Hello, PhD, I also have a question. Is there any difference between your rnpcanet and the PCA process in pcanet? The PCA in rnpcanet has a reduced dimension, but the size remains the same. The PCA in pcanet has an increased dimension, but the size decreases

**Comment by mrmq9** [comment]
> 

Do you mean if padding is done in PCANet and RNPCANet, so the size of images doesn't change after the PCA stage?

**Comment by szj-github** [comment]
![image](https://user-images.githubusercontent.com/54060024/173341939-8538a805-3f48-4245-9222-30366f23f9dd.png)
Hello, the above figure shows the pcanet structure. Taking the first stage as an example, X is convoluted by L1 different filters, and finally rises to the L1 dimension. Therefore, the PCA stage in the pcanet is equivalent to a dimension rising process, right?

**Comment by mrmq9** [comment]
> phd，I wonder if you use Skype app
Yes, can you give me your email or mail me? you can find my email in the paper
